tests/ui/anonymous-higher-ranked-lifetime.stderr

```diff
@@ -2,8 +2,9 @@ error[E0631]: type mismatch in closure arguments
   --> $DIR/anonymous-higher-ranked-lifetime.rs:2:5
    |
 LL |     f1(|_: (), _: ()| {});
-   |     ^^ -------------- found signature defined here
-   |     |
+   |     ^^^--------------^^^^
```

Comment:
There are lots of new occurrences of `^^^---^^^` now. I guess the `^^^` highlighting and the `---` highlighting overlap, and the `---` wins?

Also, it's weird that the `f1` and the `{}` are both pointed to by the `^` markers, they don't really seem connected?

Reply:
There's some logic so when primary and secondary spans overlap some bit of information is shown: the largest one is printed "first", and the then each shorter one is printed in order of length, so that at least a bit of each is shown. This is because if, for example in this case, there is a span with a label pointing within a larger one and we *weren't* doing that, we'd have no way of knowing where the label span ended within the larger span. That's how you end up with this kind of rendering. IIRC, we also tried some tricks with colors to aid here, but don't think they panned out so we didn't land them.

In this case the primary span marks from the `fq` all the way to the closing `)` (including the `{}`), it's just "visually interrupted" by the span pointing at the closure header.

Reply:
Makes sense. Not for this PR, but I wonder if changing this:
```
LL |     f1(|_: (), _: ()| {});
   |     ^^^--------------^^^^
   |     |  |
   |     |  found signature defined here
   |     expected due to this
```
to this:
```
LL |     f1(|_: (), _: ()| {});
   |     ^^^^^^^^^^^^^^^^^^^^^
   |     |  --------------
   |     |  |
   |     |  found signature defined here
   |     expected due to this
```
would help.

W.r.t. my lack of understanding of `---` marking the secondary span, I think it's a lot clearer that `---` is marking code when the secondary span label is underneath the `---` and there's a `|` leading to it, instead of the label being to the right on the same line.